vignettes/rebuild_cached_data.Rmd

```diff
@@ -1,128 +1,110 @@
-save(cadlb, file = "data/cadlb.RData", compress = "xz")
-save(cadmh, file = "data/cadmh.RData", compress = "xz")
-save(cadpc, file = "data/cadpc.RData", compress = "xz")
-save(cadpp, file = "data/cadpp.RData", compress = "xz")
-save(cadqs, file = "data/cadqs.RData", compress = "xz")
-save(cadrs, file = "data/cadrs.RData", compress = "xz")
-save(cadtr, file = "data/cadtr.RData", compress = "xz")
-save(cadtte, file = "data/cadtte.RData", compress = "xz")
-save(cadvs, file = "data/cadvs.RData", compress = "xz")
-save(caddv, file = "data/caddv.RData", compress = "xz")
-save(cadsub, file = "data/cadsub.RData", compress = "xz")
-save(cadhy, file = "data/cadhy.RData", compress = "xz")
-save(cadqlqc, file = "data/cadqlqc.RData", compress = "xz")
+# Datasets must be listed after all of their dependencies
+# e.g. adsl is a dependency for all other datasets so it is listed first.
+pkg_dir <- dirname(getwd())
+
+# Listing source files and extraction of datasets' names
+src_files <- list.files(paste0(pkg_dir, "/R"))
+data_nms <- src_files[grepl("^ra*", src_files)] %>% 
+  stringr::str_remove(pattern = "^r") %>% 
+  stringr::str_remove(pattern = ".R$") %>% 
+  sort() 
+
+# Exception handling
+data_nms <- data_nms[data_nms != "adsaftte"] # Unbuilt for now
+
+# Construction of dependency tree based on formals
+data_deps <- sapply(
+  data_nms,
+  function(x) {
+    dat_args <- names(formals(paste0("r", x)))
+    dat_args[dat_args %in% data_nms]
+  }
+)
+
+updated_files <- system("git diff origin/main --name-only", intern = TRUE)
+updated_data <- updated_files[grepl("^R\\/", updated_files)] %>% 
+  stringr::str_remove("^R\\/") %>% 
+  stringr::str_remove(pattern = "^r") %>% 
+  stringr::str_remove(pattern = ".R$") 
+
+if (length(updated_data) != 0) {
+  # Extracting relative dependencies
+  update_after_deps <- unlist(
+    sapply(seq_along(data_deps), function(x) {
+      if (updated_data %in% data_deps[[x]]) {
+        names(data_deps)[x]
+      }
+    })
+  )
+  data_to_update <- unique(c(updated_data, update_after_deps))
+  data_to_update <- data_to_update[order(match(data_to_update, data_nms))] # order dependencies first
+  
+  # Generate and save updated cached datasets
+  default_args <- list(seed = 1, na_vars = list(), who_coding = TRUE, percent = 80, number = 2)
+  for (dat in data_to_update) {
+    # Match arguments with defaults
+    dat_args <- default_args[names(default_args) %in% names(formals(paste0("r", dat)))]
+    
+    # Get the data deps cache that is already there (if adsl returns list())
+    dat_deps <- lapply(data_deps[[dat]], function(x) get(paste0("c", x)))
+    
+    # Main call to creation function
+    cdataset <- do.call(paste0("r", dat), c(dat_args, dat_deps))
+    attr(cdataset, "creation date") <- lubridate::date() # nolint
+  
+    # Preview differences (optional)
+    cat("\nSaving cache for dataset", dat, "with following changes (diffdf):\n")
+    print(diffdf(get(paste0("c", dat)), cdataset))
```

Comment:
@edelarua why do you think I get the following?
```
Saving cache for dataset adab with the following changes (diffdf):
No issues were found!
Dataset cache updated for adab in file cadab.RData 
```

Reply:
Can you try to remove the data update and see if prints the fiff?

Reply:
Since nothing was actually changed in `radab` that would affect the dataset values that's the expected result.